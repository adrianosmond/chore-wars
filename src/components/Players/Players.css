.players {
  display: flex;
  justify-content: space-around;
  width: 100%;
}

.players__player {
  width: 200px;
}

.players__avatar {
  display: block;
  position: relative;
}

.players__player-name {
  display: block;
  font-weight: bold;
  line-height: 1;
  text-align: center;
  margin-top: 0.5rem;
  margin-bottom: 0.5rem;
  text-decoration: none;
  color: inherit;
}

.players__paid-button {
  position: absolute;
  bottom: 2%;
}

.players__paid-button--1 {
  left: 20%;
}

.players__paid-button--2 {
  right: 20%;
}

.players__player-owed {
  font-size: 2rem;
  line-height: 1;
}

.players__owed-multiple {
  position: absolute;
  width: 2rem;
  top: 0;
  left: 0;
  line-height: 1.75rem;
  font-size: 0.5rem;
  text-align: center;
}

@media screen and (min-width: 37.5em) {
  .players__player-name {
    font-size: 1.5rem;
  }
}
