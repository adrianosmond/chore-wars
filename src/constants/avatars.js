export const avatarOptions = {
  skinColor: [
    'Pale',
    'Light',
    'Tanned',
    'Yellow',
    'Brown',
    'DarkBrown',
    'Black',
  ],
  topType: [
    'NoHair',
    'Eyepatch',
    'Hat',
    'Hijab',
    'Turban',
    'WinterHat1',
    'WinterHat2',
    'WinterHat3',
    'WinterHat4',
    'LongHairBigHair',
    'LongHairBob',
    'LongHairBun',
    'LongHairCurly',
    'LongHairCurvy',
    'LongHairDreads',
    'LongHairFrida',
    'LongHairFro',
    'LongHairFroBand',
    'LongHairNotTooLong',
    'LongHairShavedSides',
    'LongHairMiaWallace',
    'LongHairStraight',
    'LongHairStraight2',
    'LongHairStraightStrand',
    'ShortHairDreads01',
    'ShortHairDreads02',
    'ShortHairFrizzle',
    'ShortHairShaggyMullet',
    'ShortHairShortCurly',
    'ShortHairShortFlat',
    'ShortHairShortRound',
    'ShortHairShortWaved',
    'ShortHairSides',
    'ShortHairTheCaesar',
    'ShortHairTheCaesarSidePart',
  ],
  hatColor: [
    'Black',
    'Blue01',
    'Blue02',
    'Blue03',
    'Gray01',
    'Gray02',
    'Heather',
    'PastelBlue',
    'PastelGreen',
    'PastelOrange',
    'PastelRed',
    'PastelYellow',
    'Pink',
    'Red',
    'White',
  ],
  hairColor: [
    'Auburn',
    'Black',
    'Blonde',
    'BlondeGolden',
    'Brown',
    'BrownDark',
    'PastelPink',
    'Platinum',
    'Red',
    'SilverGray',
  ],
  facialHairType: [
    'Blank',
    'BeardMedium',
    'BeardLight',
    'BeardMagestic',
    'MoustacheFancy',
    'MoustacheMagnum',
  ],
  facialHairColor: [
    'Auburn',
    'Black',
    'Blonde',
    'BlondeGolden',
    'Brown',
    'BrownDark',
    'Platinum',
    'Red',
  ],
  accessoriesType: [
    'Blank',
    'Kurt',
    'Prescription01',
    'Prescription02',
    'Round',
    'Sunglasses',
    'Wayfarers',
  ],
  clotheType: [
    'BlazerShirt',
    'BlazerSweater',
    'CollarSweater',
    'Hoodie',
    'Overall',
    'ShirtCrewNeck',
    'ShirtScoopNeck',
    'ShirtVNeck',
  ],
  clotheColor: [
    'Black',
    'Blue01',
    'Blue02',
    'Blue03',
    'Gray01',
    'Gray02',
    'Heather',
    'PastelBlue',
    'PastelGreen',
    'PastelOrange',
    'PastelRed',
    'PastelYellow',
    'Pink',
    'Red',
    'White',
  ],
};


export const editorOrder = [
  'skinColor',
  'topType',
  'hairColor',
  'hatColor',
  'facialHairType',
  'facialHairColor',
  'accessoriesType',
  'clotheType',
  'clotheColor',
];

export const labels = {
  skinColor: 'Skin Colour',
  topType: 'Hair or Headware',
  hairColor: 'Hair Colour',
  hatColor: 'Headware Colour',
  facialHairType: 'Facial Hair Type',
  facialHairColor: 'Facial Hair Colour',
  accessoriesType: 'Glasses',
  clotheType: 'Clothing Type',
  clotheColor: 'Clothing Colour',
};

export const isEditable = {
  skinColor: () => true,
  topType: () => true,
  hairColor: options => [
    'NoHair',
    'Eyepatch',
    'Hat',
    'Hijab',
    'Turban',
    'WinterHat1',
    'WinterHat2',
    'WinterHat3',
    'WinterHat4',
    'LongHairFrida',
    'LongHairShavedSides',
  ].indexOf(options.topType) < 0,
  hatColor: options => [
    'Hijab',
    'Turban',
    'WinterHat1',
    'WinterHat2',
    'WinterHat3',
    'WinterHat4',
  ].indexOf(options.topType) >= 0,
  facialHairType: options => options.topType !== 'Hijab',
  facialHairColor: options => isEditable.facialHairType(options) && options.facialHairType !== 'Blank',
  accessoriesType: () => true,
  clotheType: () => true,
  clotheColor: options => !options.clotheType.startsWith('Blazer'),
};

export const DefaultAvatar = {
  accessoriesType: 'Blank',
  avatarStyle: 'Transparent',
  clotheColor: 'Blue03',
  clotheType: 'BlazerShirt',
  eyeType: 'Default',
  eyebrowType: 'Default',
  facialHairColor: 'BrownDark',
  facialHairType: 'Blank',
  hairColor: 'BrownDark',
  mouthType: 'Twinkle',
  skinColor: 'Light',
  topType: 'LongHairStraight',
};
