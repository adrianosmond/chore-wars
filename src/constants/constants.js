const TIME_UNIT = 1000 * 60 * 60 * 24; // 1 day
const MIN_CHORE_FREQUENCY = 1;
const MAX_CHORE_FREQUENCY = 365;
const MIN_CHORE_POINTS = 1;
const MAX_CHORE_POINTS = 200;

const DATE_FORMAT = 'DD/MM/YYYY';

const MAX_POINT_DIFFERENCE = 500;

const MAX_NAME_LENGTH = 10;
const JOIN_CODE_LENGTH = 8;

const DefaultAvatar = {
  accessoriesType: 'Blank',
  avatarStyle: 'Transparent',
  clotheColor: 'Blue03',
  clotheType: 'BlazerShirt',
  eyeType: 'Default',
  eyebrowType: 'Default',
  facialHairColor: 'BrownDark',
  facialHairType: 'Blank',
  hairColor: 'BrownDark',
  mouthType: 'Twinkle',
  skinColor: 'Light',
  topType: 'LongHairStraight',
};

const ActionTypes = {
  addChore: 'ADD_CHORE',
  resetChoreDoneDate: 'RESET_CHORE_DONE_DATE',
  blockChore: 'BLOCK_CHORE',
  unblockChore: 'UNBLOCK_CHORE',
  removeChore: 'REMOVE_CHORE',
  updateChore: 'UPDATE_CHORE',
  setChores: 'SET_CHORES',
  makeChain: 'MAKE_CHAIN',
  breakChain: 'BREAK_CHAIN',
  addPoints: 'ADD_POINTS',
  claimPrize: 'CLAIM_PRIZE',
  paidDebt: 'PAID_DEBT',
  setPoints: 'SET_POINTS',
  updateUser: 'UPDATE_USER',
  setAuthUser: 'SET_AUTH_USER',
  setGame: 'SET_GAME',
  setPointsLoaded: 'SET_POINTS_LOADED',
  setChoresLoaded: 'SET_CHORES_LOADED',
  saveStatePostUndo: 'SAVE_STATE_POST_UNDO',
};

export {
  TIME_UNIT,
  MIN_CHORE_FREQUENCY,
  MAX_CHORE_FREQUENCY,
  MIN_CHORE_POINTS,
  MAX_CHORE_POINTS,
  MAX_NAME_LENGTH,
  JOIN_CODE_LENGTH,
  DATE_FORMAT,
  MAX_POINT_DIFFERENCE,
  ActionTypes,
  DefaultAvatar,
};
