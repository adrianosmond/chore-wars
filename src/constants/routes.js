export const LOGIN = '/'
export const CHORES = '/chores'
export const NEW_CHORE = '/chores/new'
