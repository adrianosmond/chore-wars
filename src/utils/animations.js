export const onAppear = (el) => {
  setTimeout(() => {
    el.classList.add('fadeIn');
    setTimeout(() => {
      // eslint-disable-next-line no-param-reassign
      el.style.opacity = '';
      el.classList.remove('fadeIn');
    }, 500);
  }, 0);
};

export const onExit = (el, _, removeElement) => {
  setTimeout(() => {
    el.classList.add('fadeOut');
    setTimeout(removeElement, 500);
  }, 0);
};
