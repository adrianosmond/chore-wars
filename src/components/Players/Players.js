import React from 'react';
import Avatar from 'avataaars';

import OwedBadge from 'components/OwedBadge';

import './Players.css';

const Players = ({ players, points, gameId }) => (
  <div className="players">
    {players.map((player, idx) => (
      <div className={`players__player players__player--${idx + 1}`} key={player.id}>
        <div className="players__avatar">
          <Avatar style={{ width: '100%', height: '100%', display: 'block' }} {...player.avatar} />
          <OwedBadge player={player} idx={idx} points={points} gameId={gameId} />
        </div>
        <div className="players__player-name">{player.name}</div>
      </div>
    ))}
    { players.length < 2 ? (
      <div className="players__player">
        <p>Invite someone to sign up and give them the code:</p>
        <p id="join-code">{players[0].joinCode}</p>
        <p>to get them to join this game.</p>
      </div>
    ) : null }
  </div>
);

export default Players;
