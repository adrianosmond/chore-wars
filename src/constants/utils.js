import { TIME_UNIT } from './constants'
const sortByCurrentPoints = (a, b) => b.currentPoints - a.currentPoints

const processChore = (chore) => {
  const timeSinceChore = (new Date().getTime() - chore.lastDone) / TIME_UNIT
  const timeRemaining = timeSinceChore - chore.frequency
  const percentage = Math.min(100 * timeSinceChore / chore.frequency, 100)
  const multiplier = 1 + (1 / chore.frequency)
  const due = chore.lastDone + (chore.frequency * TIME_UNIT)
  return {
    currentPoints: Math.round(percentage / 100 * chore.pointsPerTime * Math.pow(multiplier, timeRemaining)),
    percentage,
    due
  }
}

export { sortByCurrentPoints, processChore }
