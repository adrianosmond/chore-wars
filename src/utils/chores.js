import { TIME_UNIT, MAX_CHORE_CURRENT_POINTS } from 'constants/constants';

export const getCurrentPoints = (chore, timeSinceChore, percentage) => {
  if (chore.frequency === 0) {
    return chore.pointsPerTime;
  }
  const multiplier = 1 + 1 / chore.frequency;
  const timeRemaining = timeSinceChore - chore.frequency;
  return Math.min(
    MAX_CHORE_CURRENT_POINTS,
    Math.round(
      (percentage / 100) * chore.pointsPerTime * multiplier ** timeRemaining,
    ),
  );
};

export const computedChoreProperties = (chore, now) => {
  const timePaused = chore.timePaused || 0;
  const timeSinceChore = (now - (chore.lastDone + timePaused)) / TIME_UNIT;
  const percentage =
    chore.frequency === 0
      ? 100
      : Math.min((100 * timeSinceChore) / chore.frequency, 100);
  const due = chore.lastDone + timePaused + chore.frequency * TIME_UNIT;
  const currentPoints = getCurrentPoints(chore, timeSinceChore, percentage);
  return {
    currentPoints,
    percentage,
    due,
  };
};

export const makeChoresArray = (chores, now = new Date().getTime()) =>
  Object.keys(chores).map(chore => ({
    ...chores[chore],
    id: chore,
    ...computedChoreProperties(chores[chore], now),
  }));

export const choreIsAvailable = chore =>
  !chore.isWaiting && new Date().getTime() - chore.lastDone >= 10000;

export const sortByCurrentPoints = (a, b) => {
  const aDue = a.currentPoints >= a.pointsPerTime;
  const bDue = b.currentPoints >= b.pointsPerTime;
  if (aDue && !bDue) {
    return -1;
  }
  if (bDue && !aDue) {
    return 1;
  }
  return b.currentPoints - a.currentPoints;
};

export const getFrequencyDescription = frequency => {
  if (frequency === 0) return 'when it needs to be done';
  if (frequency === 1) return 'every day';
  return `every ${frequency} days`;
};
