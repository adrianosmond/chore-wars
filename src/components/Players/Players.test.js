import React from 'react';
import { shallow } from 'enzyme';
import { makePlayersArray } from 'constants/utils';
import { DefaultAvatar } from 'constants/avatars';
import Players from './Players';

describe('Players', () => {
  it('Renders players', () => {
    const playerPoints = {
      player1: {
        avatar: DefaultAvatar,
        name: 'Player 1',
      },
      player2: {
        avatar: DefaultAvatar,
        name: 'Player 2',
      },
    };
    const players = makePlayersArray(playerPoints);
    const playersComponent = shallow(
      <Players
        players={players}
        points={{}}
        gameId="test-game"
      />,
    );
    expect(playersComponent).toMatchSnapshot();
  });

  describe('With players being owed', () => {
    it('Renders players with p1 being owed', () => {
      const playerPoints = {
        player1: {
          avatar: DefaultAvatar,
          name: 'Player 1',
        },
        player2: {
          avatar: DefaultAvatar,
          name: 'Player 2',
        },
      };
      const players = makePlayersArray(playerPoints);
      const playersComponent = shallow(
        <Players
          players={players}
          points={{}}
          gameId="test-game"
        />,
      );
      expect(playersComponent).toMatchSnapshot();
    });

    it('Renders players with p2 being owed', () => {
      const playerPoints = {
        player1: {
          avatar: DefaultAvatar,
          name: 'Player 1',
        },
        player2: {
          avatar: DefaultAvatar,
          name: 'Player 2',
        },
      };
      const players = makePlayersArray(playerPoints);
      const playersComponent = shallow(
        <Players
          players={players}
          points={{}}
          gameId="test-game"
        />,
      );
      expect(playersComponent).toMatchSnapshot();
    });

    it('Renders players with p1 being owed > 1', () => {
      const playerPoints = {
        player1: {
          avatar: DefaultAvatar,
          name: 'Player 1',
        },
        player2: {
          avatar: DefaultAvatar,
          name: 'Player 2',
        },
      };
      const players = makePlayersArray(playerPoints);
      const playersComponent = shallow(
        <Players
          players={players}
          points={{}}
          gameId="test-game"
        />,
      );
      expect(playersComponent).toMatchSnapshot();
    });

    it('Renders players with p2 being owed > 1', () => {
      const playerPoints = {
        player1: {
          avatar: DefaultAvatar,
          name: 'Player 1',
        },
        player2: {
          avatar: DefaultAvatar,
          name: 'Player 2',
        },
      };
      const players = makePlayersArray(playerPoints);
      const playersComponent = shallow(
        <Players
          players={players}
          points={{}}
          gameId="test-game"
        />,
      );
      expect(playersComponent).toMatchSnapshot();
    });
  });

  describe('With one player', () => {
    it('Renders and displays the join code', () => {
      const joinCode = 'abcdefgh';
      const playerPoints = {
        player1: {
          avatar: DefaultAvatar,
          name: 'Player 1',
          joinCode,
        },
      };
      const players = makePlayersArray(playerPoints);
      const playersComponent = shallow(
        <Players
          players={players}
          points={{}}
          gameId="test-game"
        />,
      );
      expect(playersComponent).toMatchSnapshot();
      expect(playersComponent.find('#join-code').text()).toBe(joinCode);
    });
  });
});
