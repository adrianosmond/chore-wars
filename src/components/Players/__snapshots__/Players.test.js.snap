// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Players Renders players 1`] = `
<div
  className="players"
>
  <div
    className="players__player players__player--1"
    key="player1"
  >
    <div
      className="players__avatar"
    >
      <AvatarComponent
        accessoriesType="Blank"
        avatarStyle="Transparent"
        clotheColor="Blue03"
        clotheType="BlazerShirt"
        eyeType="Default"
        eyebrowType="Default"
        facialHairColor="BrownDark"
        facialHairType="Blank"
        hairColor="BrownDark"
        mouthType="Twinkle"
        skinColor="Light"
        style={
          Object {
            "display": "block",
            "height": "100%",
            "width": "100%",
          }
        }
        topType="LongHairStraight"
      />
      <OwedBadge
        gameId="test-game"
        idx={0}
        player={
          Object {
            "avatar": Object {
              "accessoriesType": "Blank",
              "avatarStyle": "Transparent",
              "clotheColor": "Blue03",
              "clotheType": "BlazerShirt",
              "eyeType": "Default",
              "eyebrowType": "Default",
              "facialHairColor": "BrownDark",
              "facialHairType": "Blank",
              "hairColor": "BrownDark",
              "mouthType": "Twinkle",
              "skinColor": "Light",
              "topType": "LongHairStraight",
            },
            "id": "player1",
            "name": "Player 1",
          }
        }
        points={Object {}}
      />
    </div>
    <div
      className="players__player-name"
    >
      Player 1
    </div>
  </div>
  <div
    className="players__player players__player--2"
    key="player2"
  >
    <div
      className="players__avatar"
    >
      <AvatarComponent
        accessoriesType="Blank"
        avatarStyle="Transparent"
        clotheColor="Blue03"
        clotheType="BlazerShirt"
        eyeType="Default"
        eyebrowType="Default"
        facialHairColor="BrownDark"
        facialHairType="Blank"
        hairColor="BrownDark"
        mouthType="Twinkle"
        skinColor="Light"
        style={
          Object {
            "display": "block",
            "height": "100%",
            "width": "100%",
          }
        }
        topType="LongHairStraight"
      />
      <OwedBadge
        gameId="test-game"
        idx={1}
        player={
          Object {
            "avatar": Object {
              "accessoriesType": "Blank",
              "avatarStyle": "Transparent",
              "clotheColor": "Blue03",
              "clotheType": "BlazerShirt",
              "eyeType": "Default",
              "eyebrowType": "Default",
              "facialHairColor": "BrownDark",
              "facialHairType": "Blank",
              "hairColor": "BrownDark",
              "mouthType": "Twinkle",
              "skinColor": "Light",
              "topType": "LongHairStraight",
            },
            "id": "player2",
            "name": "Player 2",
          }
        }
        points={Object {}}
      />
    </div>
    <div
      className="players__player-name"
    >
      Player 2
    </div>
  </div>
</div>
`;

exports[`Players With one player Renders and displays the join code 1`] = `
<div
  className="players"
>
  <div
    className="players__player players__player--1"
    key="player1"
  >
    <div
      className="players__avatar"
    >
      <AvatarComponent
        accessoriesType="Blank"
        avatarStyle="Transparent"
        clotheColor="Blue03"
        clotheType="BlazerShirt"
        eyeType="Default"
        eyebrowType="Default"
        facialHairColor="BrownDark"
        facialHairType="Blank"
        hairColor="BrownDark"
        mouthType="Twinkle"
        skinColor="Light"
        style={
          Object {
            "display": "block",
            "height": "100%",
            "width": "100%",
          }
        }
        topType="LongHairStraight"
      />
      <OwedBadge
        gameId="test-game"
        idx={0}
        player={
          Object {
            "avatar": Object {
              "accessoriesType": "Blank",
              "avatarStyle": "Transparent",
              "clotheColor": "Blue03",
              "clotheType": "BlazerShirt",
              "eyeType": "Default",
              "eyebrowType": "Default",
              "facialHairColor": "BrownDark",
              "facialHairType": "Blank",
              "hairColor": "BrownDark",
              "mouthType": "Twinkle",
              "skinColor": "Light",
              "topType": "LongHairStraight",
            },
            "id": "player1",
            "joinCode": "abcdefgh",
            "name": "Player 1",
          }
        }
        points={Object {}}
      />
    </div>
    <div
      className="players__player-name"
    >
      Player 1
    </div>
  </div>
  <div
    className="players__player"
  >
    <p>
      Invite someone to sign up and give them the code:
    </p>
    <p
      id="join-code"
    >
      abcdefgh
    </p>
    <p>
      to get them to join this game.
    </p>
  </div>
</div>
`;

exports[`Players With players being owed Renders players with p1 being owed > 1 1`] = `
<div
  className="players"
>
  <div
    className="players__player players__player--1"
    key="player1"
  >
    <div
      className="players__avatar"
    >
      <AvatarComponent
        accessoriesType="Blank"
        avatarStyle="Transparent"
        clotheColor="Blue03"
        clotheType="BlazerShirt"
        eyeType="Default"
        eyebrowType="Default"
        facialHairColor="BrownDark"
        facialHairType="Blank"
        hairColor="BrownDark"
        mouthType="Twinkle"
        skinColor="Light"
        style={
          Object {
            "display": "block",
            "height": "100%",
            "width": "100%",
          }
        }
        topType="LongHairStraight"
      />
      <OwedBadge
        gameId="test-game"
        idx={0}
        player={
          Object {
            "avatar": Object {
              "accessoriesType": "Blank",
              "avatarStyle": "Transparent",
              "clotheColor": "Blue03",
              "clotheType": "BlazerShirt",
              "eyeType": "Default",
              "eyebrowType": "Default",
              "facialHairColor": "BrownDark",
              "facialHairType": "Blank",
              "hairColor": "BrownDark",
              "mouthType": "Twinkle",
              "skinColor": "Light",
              "topType": "LongHairStraight",
            },
            "id": "player1",
            "name": "Player 1",
          }
        }
        points={Object {}}
      />
    </div>
    <div
      className="players__player-name"
    >
      Player 1
    </div>
  </div>
  <div
    className="players__player players__player--2"
    key="player2"
  >
    <div
      className="players__avatar"
    >
      <AvatarComponent
        accessoriesType="Blank"
        avatarStyle="Transparent"
        clotheColor="Blue03"
        clotheType="BlazerShirt"
        eyeType="Default"
        eyebrowType="Default"
        facialHairColor="BrownDark"
        facialHairType="Blank"
        hairColor="BrownDark"
        mouthType="Twinkle"
        skinColor="Light"
        style={
          Object {
            "display": "block",
            "height": "100%",
            "width": "100%",
          }
        }
        topType="LongHairStraight"
      />
      <OwedBadge
        gameId="test-game"
        idx={1}
        player={
          Object {
            "avatar": Object {
              "accessoriesType": "Blank",
              "avatarStyle": "Transparent",
              "clotheColor": "Blue03",
              "clotheType": "BlazerShirt",
              "eyeType": "Default",
              "eyebrowType": "Default",
              "facialHairColor": "BrownDark",
              "facialHairType": "Blank",
              "hairColor": "BrownDark",
              "mouthType": "Twinkle",
              "skinColor": "Light",
              "topType": "LongHairStraight",
            },
            "id": "player2",
            "name": "Player 2",
          }
        }
        points={Object {}}
      />
    </div>
    <div
      className="players__player-name"
    >
      Player 2
    </div>
  </div>
</div>
`;

exports[`Players With players being owed Renders players with p1 being owed 1`] = `
<div
  className="players"
>
  <div
    className="players__player players__player--1"
    key="player1"
  >
    <div
      className="players__avatar"
    >
      <AvatarComponent
        accessoriesType="Blank"
        avatarStyle="Transparent"
        clotheColor="Blue03"
        clotheType="BlazerShirt"
        eyeType="Default"
        eyebrowType="Default"
        facialHairColor="BrownDark"
        facialHairType="Blank"
        hairColor="BrownDark"
        mouthType="Twinkle"
        skinColor="Light"
        style={
          Object {
            "display": "block",
            "height": "100%",
            "width": "100%",
          }
        }
        topType="LongHairStraight"
      />
      <OwedBadge
        gameId="test-game"
        idx={0}
        player={
          Object {
            "avatar": Object {
              "accessoriesType": "Blank",
              "avatarStyle": "Transparent",
              "clotheColor": "Blue03",
              "clotheType": "BlazerShirt",
              "eyeType": "Default",
              "eyebrowType": "Default",
              "facialHairColor": "BrownDark",
              "facialHairType": "Blank",
              "hairColor": "BrownDark",
              "mouthType": "Twinkle",
              "skinColor": "Light",
              "topType": "LongHairStraight",
            },
            "id": "player1",
            "name": "Player 1",
          }
        }
        points={Object {}}
      />
    </div>
    <div
      className="players__player-name"
    >
      Player 1
    </div>
  </div>
  <div
    className="players__player players__player--2"
    key="player2"
  >
    <div
      className="players__avatar"
    >
      <AvatarComponent
        accessoriesType="Blank"
        avatarStyle="Transparent"
        clotheColor="Blue03"
        clotheType="BlazerShirt"
        eyeType="Default"
        eyebrowType="Default"
        facialHairColor="BrownDark"
        facialHairType="Blank"
        hairColor="BrownDark"
        mouthType="Twinkle"
        skinColor="Light"
        style={
          Object {
            "display": "block",
            "height": "100%",
            "width": "100%",
          }
        }
        topType="LongHairStraight"
      />
      <OwedBadge
        gameId="test-game"
        idx={1}
        player={
          Object {
            "avatar": Object {
              "accessoriesType": "Blank",
              "avatarStyle": "Transparent",
              "clotheColor": "Blue03",
              "clotheType": "BlazerShirt",
              "eyeType": "Default",
              "eyebrowType": "Default",
              "facialHairColor": "BrownDark",
              "facialHairType": "Blank",
              "hairColor": "BrownDark",
              "mouthType": "Twinkle",
              "skinColor": "Light",
              "topType": "LongHairStraight",
            },
            "id": "player2",
            "name": "Player 2",
          }
        }
        points={Object {}}
      />
    </div>
    <div
      className="players__player-name"
    >
      Player 2
    </div>
  </div>
</div>
`;

exports[`Players With players being owed Renders players with p2 being owed > 1 1`] = `
<div
  className="players"
>
  <div
    className="players__player players__player--1"
    key="player1"
  >
    <div
      className="players__avatar"
    >
      <AvatarComponent
        accessoriesType="Blank"
        avatarStyle="Transparent"
        clotheColor="Blue03"
        clotheType="BlazerShirt"
        eyeType="Default"
        eyebrowType="Default"
        facialHairColor="BrownDark"
        facialHairType="Blank"
        hairColor="BrownDark"
        mouthType="Twinkle"
        skinColor="Light"
        style={
          Object {
            "display": "block",
            "height": "100%",
            "width": "100%",
          }
        }
        topType="LongHairStraight"
      />
      <OwedBadge
        gameId="test-game"
        idx={0}
        player={
          Object {
            "avatar": Object {
              "accessoriesType": "Blank",
              "avatarStyle": "Transparent",
              "clotheColor": "Blue03",
              "clotheType": "BlazerShirt",
              "eyeType": "Default",
              "eyebrowType": "Default",
              "facialHairColor": "BrownDark",
              "facialHairType": "Blank",
              "hairColor": "BrownDark",
              "mouthType": "Twinkle",
              "skinColor": "Light",
              "topType": "LongHairStraight",
            },
            "id": "player1",
            "name": "Player 1",
          }
        }
        points={Object {}}
      />
    </div>
    <div
      className="players__player-name"
    >
      Player 1
    </div>
  </div>
  <div
    className="players__player players__player--2"
    key="player2"
  >
    <div
      className="players__avatar"
    >
      <AvatarComponent
        accessoriesType="Blank"
        avatarStyle="Transparent"
        clotheColor="Blue03"
        clotheType="BlazerShirt"
        eyeType="Default"
        eyebrowType="Default"
        facialHairColor="BrownDark"
        facialHairType="Blank"
        hairColor="BrownDark"
        mouthType="Twinkle"
        skinColor="Light"
        style={
          Object {
            "display": "block",
            "height": "100%",
            "width": "100%",
          }
        }
        topType="LongHairStraight"
      />
      <OwedBadge
        gameId="test-game"
        idx={1}
        player={
          Object {
            "avatar": Object {
              "accessoriesType": "Blank",
              "avatarStyle": "Transparent",
              "clotheColor": "Blue03",
              "clotheType": "BlazerShirt",
              "eyeType": "Default",
              "eyebrowType": "Default",
              "facialHairColor": "BrownDark",
              "facialHairType": "Blank",
              "hairColor": "BrownDark",
              "mouthType": "Twinkle",
              "skinColor": "Light",
              "topType": "LongHairStraight",
            },
            "id": "player2",
            "name": "Player 2",
          }
        }
        points={Object {}}
      />
    </div>
    <div
      className="players__player-name"
    >
      Player 2
    </div>
  </div>
</div>
`;

exports[`Players With players being owed Renders players with p2 being owed 1`] = `
<div
  className="players"
>
  <div
    className="players__player players__player--1"
    key="player1"
  >
    <div
      className="players__avatar"
    >
      <AvatarComponent
        accessoriesType="Blank"
        avatarStyle="Transparent"
        clotheColor="Blue03"
        clotheType="BlazerShirt"
        eyeType="Default"
        eyebrowType="Default"
        facialHairColor="BrownDark"
        facialHairType="Blank"
        hairColor="BrownDark"
        mouthType="Twinkle"
        skinColor="Light"
        style={
          Object {
            "display": "block",
            "height": "100%",
            "width": "100%",
          }
        }
        topType="LongHairStraight"
      />
      <OwedBadge
        gameId="test-game"
        idx={0}
        player={
          Object {
            "avatar": Object {
              "accessoriesType": "Blank",
              "avatarStyle": "Transparent",
              "clotheColor": "Blue03",
              "clotheType": "BlazerShirt",
              "eyeType": "Default",
              "eyebrowType": "Default",
              "facialHairColor": "BrownDark",
              "facialHairType": "Blank",
              "hairColor": "BrownDark",
              "mouthType": "Twinkle",
              "skinColor": "Light",
              "topType": "LongHairStraight",
            },
            "id": "player1",
            "name": "Player 1",
          }
        }
        points={Object {}}
      />
    </div>
    <div
      className="players__player-name"
    >
      Player 1
    </div>
  </div>
  <div
    className="players__player players__player--2"
    key="player2"
  >
    <div
      className="players__avatar"
    >
      <AvatarComponent
        accessoriesType="Blank"
        avatarStyle="Transparent"
        clotheColor="Blue03"
        clotheType="BlazerShirt"
        eyeType="Default"
        eyebrowType="Default"
        facialHairColor="BrownDark"
        facialHairType="Blank"
        hairColor="BrownDark"
        mouthType="Twinkle"
        skinColor="Light"
        style={
          Object {
            "display": "block",
            "height": "100%",
            "width": "100%",
          }
        }
        topType="LongHairStraight"
      />
      <OwedBadge
        gameId="test-game"
        idx={1}
        player={
          Object {
            "avatar": Object {
              "accessoriesType": "Blank",
              "avatarStyle": "Transparent",
              "clotheColor": "Blue03",
              "clotheType": "BlazerShirt",
              "eyeType": "Default",
              "eyebrowType": "Default",
              "facialHairColor": "BrownDark",
              "facialHairType": "Blank",
              "hairColor": "BrownDark",
              "mouthType": "Twinkle",
              "skinColor": "Light",
              "topType": "LongHairStraight",
            },
            "id": "player2",
            "name": "Player 2",
          }
        }
        points={Object {}}
      />
    </div>
    <div
      className="players__player-name"
    >
      Player 2
    </div>
  </div>
</div>
`;
